broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -533,17 +547,27 @@ public void onActorStarting() {
                 this.logStream = log;
                 atomixRaftPartition.getServer().addCommitListener(this);
                 atomixRaftPartition.addRoleChangeListener(this);
+                onRecovered();
                 onRoleChange(atomixRaftPartition.getRole(), atomixRaftPartition.term());
               } else {
                 LOG.error(
                     "Failed to install log stream service for partition {}",
                     atomixRaftPartition.id().id(),
                     error);
                 actor.close();
+                onFailure();
               }
             });
   }
 
+  @Override
+  protected void onActorStarted() {
+    criticalComponentsHealthMonitor.startMonitoring();
+    criticalComponentsHealthMonitor.addFailureListener(
+        FailureListener.withListeners(
```

Comment:
Can you elaborate on why the partition itself does not implement `FailureListener`?

Reply:
That is actually better. Will update.